airflow/bin/cli.py

```diff
@@ -519,25 +500,7 @@ def run(args, dag=None):
         if os.path.exists(args.cfg_path):
             os.remove(args.cfg_path)
 
-        # Do not log these properties since some may contain passwords.
-        # This may also set default values for database properties like
-        # core.sql_alchemy_pool_size
-        # core.sql_alchemy_pool_recycle
-        for section, config in conf_dict.items():
-            for option, value in config.items():
-                try:
-                    conf.set(section, option, value)
-                except NoSectionError:
-                    no_section_msg = (
-                        'Section {section} Option {option} '
-                        'does not exist in the config!'
-                    ).format(section=section, option=option)
-
-                    if section in OPTIONAL_AIRFLOW_CFG_SECTIONS:
-                        log.debug(no_section_msg)
-                    else:
-                        log.error(no_section_msg)
-
+        conf.conf.read_dict(conf_dict, source=conf_file)
```

Comment:
Should `conf._validate()` be called here in case an option is overwritten that invalidates the config?

Reply:
Also, is this `source=conf_file` intentional? If so I believe it needs to be in the `with  .... as conf_file` context manager above.

Reply:
This was meant to be the source file _name_ -  so should be args.cfg_path in this case. Good spot.

Reply:
> Should `conf.conf._validate()` be called here in case an option is overwritten that invalidates the config?

Done, by overriding read_dict in our class.